cmd/thanos/query.go

```diff
@@ -330,21 +347,26 @@ func runQuery(
 		Help: "The number of times a duplicated store addresses is detected from the different configs in query",
 	})
 
-	dialOpts, err := extgrpc.StoreClientGRPCOpts(logger, reg, tracer, secure, skipVerify, cert, key, caCert, serverName)
-	if err != nil {
-		return errors.Wrap(err, "building gRPC client")
+	// TLSConfig for endpoints provided in --endpoint, --endpoint.sd-files and --endpoint-strict.
+	var TLSConfig store.TLSConfiguration
+	if secure && len(endpointConfigYAML) == 0 {
```

Comment:
It's kind of confusing on 2 levels:

Users: "my flag is overridden without my knowledge" 
Code - secure is global but endpoints TLS and per config

Reply:
Addressed. 👍